core/cc/assert.h

```diff
@@ -17,8 +17,10 @@
 #ifndef CORE_ASSERT_H
 #define CORE_ASSERT_H
 
+#include <cassert>
 #include "log.h"
 
+#ifndef NDEBUG
```

Comment:
You can't change the semantics of GAPID_ASSERT unless you are 100% sure no-one relies on it NOT getting elided in release.

I would instead create a GAPID_DEBUG_ASSERT which **very clearly** denotes that it only exists in debug.
(this breaks arena::allocate_aligned for example).